SYCL/DeviceLib/built-ins/marray_math.cpp

```diff
@@ -0,0 +1,77 @@
+        deviceQueue.submit([&](sycl::handler &cgh) {                           \
+          sycl::accessor res_access{b, cgh};                                   \
+          cgh.single_task([=]() {                                              \
+            sycl::marray<MARRAY_ELEM_TYPE, DIM> res = FUNC(__VA_ARGS__);       \
+            for (int i = 0; i < DIM; i++)                                      \
+              res_access[i] = res[i];                                          \
+          });                                                                  \
+        });                                                                    \
+      }                                                                        \
+      for (int i = 0; i < DIM; i++)                                            \
+        assert(abs(result[i] - EXPECTED[i]) <= DELTA);                         \
+    }                                                                          \
+  }
+
+#define TEST2(FUNC, MARRAY_ELEM_TYPE, DIM, EXPECTED, DELTA, ...)               \
+  {                                                                            \
+    {                                                                          \
+      int result[DIM];                                                         \
+      {                                                                        \
+        sycl::buffer<int> b(result, sycl::range{DIM});                         \
+        deviceQueue.submit([&](sycl::handler &cgh) {                           \
+          sycl::accessor res_access{b, cgh};                                   \
+          cgh.single_task([=]() {                                              \
+            sycl::marray<int, DIM> res = FUNC(__VA_ARGS__);                    \
+            for (int i = 0; i < DIM; i++)                                      \
+              res_access[i] = res[i];                                          \
+          });                                                                  \
+        });                                                                    \
+      }                                                                        \
+      for (int i = 0; i < DIM; i++)                                            \
+        assert(std::abs(result[i] - EXPECTED[i]) <= DELTA);                    \
+    }                                                                          \
+  }
+
+#define EXPECTED(TYPE, ...) ((TYPE[]){__VA_ARGS__})
+
+int main() {
+  sycl::queue deviceQueue;
+
+  sycl::marray<float, 2> ma1{1.0f, 2.0f};
+  sycl::marray<float, 2> ma2{3.0f, 2.0f};
+  sycl::marray<float, 3> ma3{180, 180, 180};
+  sycl::marray<int, 3> ma4{1, 1, 1};
+  sycl::marray<float, 3> ma5{180, -180, -180};
+
+  TEST(sycl::fabs, float, 3, EXPECTED(float, 180, 180, 180), 0, ma5);
+  TEST2(sycl::ilogb, float, 3, EXPECTED(int, 7, 7, 7), 0, ma3);
```

Comment:
Unless I'm missing something, this seems simpler:
```suggestion
  TEST(sycl::ilogb, int, 3, EXPECTED(int, 7, 7, 7), 0, ma3);
```

Then we can remove `TEST2`. Alternatively `TEST2` could be defined using `TEST` to avoid code duplication.

Reply:
`ilogb` takes floats and returns ints, so this is not the usual case where function takes T type and returns T type, so we can't do this change. However, TEST2 can be merged to TEST by adding one new argument to the TEST - the type of return value